jupyterhub/apihandlers/auth.py

```diff
@@ -1,25 +1,23 @@
 """Authorization handlers"""
-
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
-
-from datetime import datetime
 import json
-from urllib.parse import (
-    parse_qsl,
-    quote,
-    urlencode,
-    urlparse,
-    urlunparse,
-)
+from datetime import datetime
```

Comment:
I'm not really a fan of the multi-line imports, but could be helpful for new contributors.

Reply:
This is from https://github.com/asottile/reorder_python_imports, and it gives its [reasons why](https://github.com/asottile/reorder_python_imports#why-this-style) to minimize git conflicts (fewer things per line means fewer chances of conflict). The two choices appear to be this and [isort](https://github.com/timothycrosley/isort/), and looking at them both, isort has one deal-breaker for me: no ability to `#noqa` a line or section for manual ordering when import order matters (https://github.com/timothycrosley/isort/issues/679).

We could also skip the import-ordering and do black-only.

Reply:
Thanks for the links @minrk. Let's keep the import-ordering as you have it. I agree that I like your choice more than isort.

Looks like the PR needs a rebase and then go ahead and merge. Thanks for doing this. I think it will keep things tidy as more folks add contributions.

Reply:
Just finished merging the outstanding PRs I wanted to before landing this one to avoid conflicts. Will rebase this one in the AM and merge.

Thanks for reviewing!